examples/graphbolt/disk_based_feature/README.md

```diff
@@ -0,0 +1,93 @@
+Training: 1178it [01:56, 10.11it/s, num_nodes=669761, gpu_cache_miss=1, cpu_cache_miss=0.0558]
+Evaluating: 123it [00:14,  8.30it/s, num_nodes=625985, gpu_cache_miss=1, cpu_cache_miss=0.0553]
+Epoch 01, Loss: 1.1435, Approx. Train: 0.6385, Approx. Val: 0.6523, Time: 116.4649817943573s
+Training: 1178it [01:58,  9.93it/s, num_nodes=663480, gpu_cache_miss=1, cpu_cache_miss=0.0478]
+Evaluating: 123it [00:14,  8.56it/s, num_nodes=624204, gpu_cache_miss=1, cpu_cache_miss=0.0476]
+Epoch 02, Loss: 1.0948, Approx. Train: 0.6516, Approx. Val: 0.6590, Time: 118.57369756698608s
+```
+
+### Performance Comparison on four caching polices
+
+Below results demonstrate the epoch time with four different caching policies.
+
+| Policy | Epoch 1 (s) | Epoch 2 (s) | Epoch 3 (s) |
+| :-----: | :---------: | :---------: | :---------: |
+|  SIEVE  |   250.589   |   116.465   |   118.574   |
+| S3-FiFO |   263.150   |   179.917   |   177.347   |
+|   LRU   |   265.154   |   157.818   |   157.937   |
+|  CLOCK  |   292.897   |   243.783   |   253.275   |
+
+## Results with Layer-Neighbor Sampling
+
+This part trains a three-layer GraphSAGE model for 3 epochs on `ogbn-papers100M` dataset with 10GB CPU cache, using Layer-Neighbor Sampling and default SIEVE policy.
+
+### Run default `--batch-dependency=1`
+
+Instruction:
+
+```
+python node_classification.py --gpu-cache-size-in-gigabytes=0 --cpu-cache-size-in-gigabytes=10 --dataset=ogbn-papers100M --sample-mode=sample_layer_neighbor --batch-dependency=1 --epochs=3
+```
+
+Result:
+
+```
+Training: 1178it [02:57,  6.63it/s, num_nodes=464607, gpu_cache_miss=1, cpu_cache_miss=0.0803]                     
+Evaluating: 123it [00:15,  7.93it/s, num_nodes=463812, gpu_cache_miss=1, cpu_cache_miss=0.0789]                    
+Epoch 00, Loss: 1.4134, Approx. Train: 0.5790, Approx. Val: 0.6277, Time: 177.58959531784058s                      
+Training: 1178it [01:34, 12.45it/s, num_nodes=476553, gpu_cache_miss=1, cpu_cache_miss=0.0617]                     
+Evaluating: 123it [00:14,  8.45it/s, num_nodes=465260, gpu_cache_miss=1, cpu_cache_miss=0.0616]                    
+Epoch 01, Loss: 1.1470, Approx. Train: 0.6383, Approx. Val: 0.6406, Time: 94.6372172832489s                        
+Training: 1178it [01:31, 12.84it/s, num_nodes=474932, gpu_cache_miss=1, cpu_cache_miss=0.0554]                     
+Evaluating: 123it [00:14,  8.70it/s, num_nodes=464145, gpu_cache_miss=1, cpu_cache_miss=0.0554]                    
+Epoch 02, Loss: 1.0990, Approx. Train: 0.6506, Approx. Val: 0.6492, Time: 91.75295209884644s
+```
+
+### Performance Comparison on different `--batch-dependency`
+
+| batch-dependency | Epoch 1 (s) | Epoch 2 (s) | Epoch 3 (s) |
+| :--------------: | :---------: | :---------: | :---------: |
+|        1        |   177.590   |   94.637   |   91.753   |
+|        64        |   174.810   |   82.209   |   81.020   |
+|       4096       |   241.959   |   76.377   |   77.082   |
```

Comment:
Why is there a discrepancy for 4096-epoch1 runtime?

Reply:
The unexpected long run time of 4096-epoch1 persists for another two runs. Papers100M has 1178 iterations an epoch, will setting the batch-dependency to be larger than that make something different?

Reply:
I need to test it myself and see I can reproduce this issue.

Reply:
It has to do with the cache. Will fix it today. Can we update these results after the cache issue is fixed?

Reply:
Sure. I will re-run the results then.

Reply:
I reran your command on my local machine with updated master.
```
   1: 97.3s, 54.4s, 53.5s
  64: 87.8s, 48.8s, 47.8s
4096: 61.1s, 31.7s, 32.2s
```
You are right, we don't actually need to make batch dependency so much greater than the number of minibatches in an epoch. Feel free to get new runtime numbers using `[1, 32, 1024]`.

Reply:
The cache implementation is fixed (hopefully), there could be significant changes to the other runtime results as well.

Reply:
I don't think you need to get these runs manually. If you implement a regression test covering all these options, we can simply copy the results from overnight results and update README.

Reply:
I will need to merge this into the master first, and then the example can be checked into the regression framework.

Reply:
#7647 fixes s3-fifo issue.

Reply:
Can we also add 1 run with 5GB GPUCache size and 20GB CPU cache size to show the users what peak speed they might be able to get? We compare this run to the 5GB GPUCache size run and 0GB GPUCache size run.

Reply:
Okay, we don't need to report the in-mem performance in this example? Or is there external statistics I can refer to?

Reply:
We should report in memory performance because we want to show if you cache enough, you will get in memory performance and the caches are not a bottleneck.

Reply:
The user is going to cache as much as their system allows anyway.

Reply:
There was a correctness issue, latest master containing #7652 should be ready for the updated results.